butterfly-utilities/src/test/java/com/paypal/butterfly/utilities/conditions/java/JavaMatchTest.java

```diff
@@ -69,6 +69,40 @@ public void noCompilationUnitTest() {
         Assert.assertEquals(executionResult.getWarnings().get(0).getMessage(), "This Java class file has no declared types: " + noCompilationUnitFile.getAbsolutePath());
     }
 
+    @Test
```

Comment:
Amrit, notice that in both of your tests the JavaMatch doesn't have any condition.

Please add a test where the JavaMatch TO has a `JavaCondition` that we would expect to return true, but, because the file is a Java package info, false and a warning are returned instead.

And a fourth test would be to the opposite, a JavaMatch with a JavaCondition that is supposed to return false, but, because the file to be evaluated is a Java package info, and because we called `setPackageInfo(true)`, true and a warning are supposed to be returned instead.

Does that make sense?

Also, if it helps, you can create a parameterized helper method containing most of the boiler plate code, and all these `@Test` methods calling it. That is optional though, your call.